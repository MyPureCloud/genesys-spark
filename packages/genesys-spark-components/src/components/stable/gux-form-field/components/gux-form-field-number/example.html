<h1>gux-form-field-number</h1>

<p>slot a input[type=number] element into a gux-form-field-number element</p>

<h2>Examples</h2>
<form onchange="notify(event)" oninput="notify(event)">
  <fieldset>
    <legend>Clearable</legend>

    <gux-form-field-number>
      <input slot="input" type="number" name="c-1" value="10" />
      <label slot="label">Default</label>
    </gux-form-field-number>

    <gux-form-field-number clearable>
      <input slot="input" type="number" name="c-2" value="10" />
      <label slot="label">True</label>
    </gux-form-field-number>

    <gux-form-field-number clearable>
      <input slot="input" type="number" name="c-3" value="10" disabled />
      <label slot="label">True and disabled</label>
    </gux-form-field-number>

    <gux-form-field-number clearable="false">
      <input slot="input" type="number" name="c-4" value="10" />
      <label slot="label">False</label>
    </gux-form-field-number>
  </fieldset>

  <fieldset>
    <legend>Label Position</legend>

    <gux-form-field-number>
      <input slot="input" type="number" name="lp-1" value="10" />
      <label slot="label">Default</label>
    </gux-form-field-number>

    <gux-form-field-number>
      <input slot="input" type="number" name="lp-2" value="10" />
      <label slot="label">Default long label</label>
    </gux-form-field-number>

    <gux-form-field-number label-position="above">
      <input slot="input" type="number" name="lp-3" value="10" />
      <label slot="label">Above</label>
    </gux-form-field-number>

    <gux-form-field-number label-position="beside">
      <input slot="input" type="number" name="lp-4" value="10" />
      <label slot="label">Beside</label>
    </gux-form-field-number>

    <gux-form-field-number label-position="screenreader">
      <input slot="input" type="number" name="lp-5" value="10" />
      <label slot="label">Screenreader</label>
    </gux-form-field-number>
  </fieldset>

  <fieldset>
    <legend>Input attributes</legend>

    <gux-form-field-number>
      <input slot="input" type="number" name="a-1" value="10" disabled />
      <label slot="label">Disabled</label>
    </gux-form-field-number>

    <gux-form-field-number>
      <input slot="input" type="number" name="a-2" value="10" required />
      <label slot="label">Required</label>
    </gux-form-field-number>

    <gux-form-field-number>
      <input slot="input" type="number" name="a-3" placeholder="Enter number" />
      <label slot="label">Placeholder</label>
    </gux-form-field-number>

    <gux-form-field-number>
      <input
        slot="input"
        type="number"
        name="a-2"
        value="10"
        step="5"
        min="-25"
        max="25"
      />
      <label slot="label">Step / Min / Max</label>
    </gux-form-field-number>
  </fieldset>

  <fieldset>
    <legend>Error</legend>

    <gux-form-field-number>
      <input slot="input" type="number" name="e-1" value="10" />
      <label slot="label">Default</label>
      <span slot="error">This is an error message</span>
    </gux-form-field-number>
  </fieldset>

  <fieldset>
    <legend>Help</legend>

    <gux-form-field-number>
      <input slot="input" type="number" name="e-1" value="10" />
      <label slot="label">Default</label>
      <span slot="help">This is a help message</span>
    </gux-form-field-number>
  </fieldset>

  <fieldset>
    <legend>Label Info</legend>

    <gux-form-field-number>
      <input slot="input" type="number" name="e-1" value="10" />
      <label slot="label">Default</label>
      <gux-label-info-beta slot="label-info">
        <span slot="content">This is some tooltip text</span>
      </gux-label-info-beta>
    </gux-form-field-number>
  </fieldset>

  <fieldset>
    <legend>Programmatically set value</legend>

    <gux-form-field-number id="p" clearable>
      <input slot="input" type="number" name="p-1" />
      <label slot="label">Random number</label>
    </gux-form-field-number>

    <gux-button-slot>
      <button
        type="button"
        onclick="(function () {
          const formField = document.getElementById('p');
          const input = document.querySelector('[name=p-1]');

          input.value = '';
          formField.guxForceUpdate();
        })()"
      >
        Clear field
      </button>
    </gux-button-slot>

    <gux-button-slot>
      <button
        type="button"
        onclick="(function () {
          const randomNumber = Math.floor(Math.random() * 10000);
          const formField = document.getElementById('p');
          const input = document.querySelector('[name=p-1]');

          input.value = randomNumber;
          formField.guxForceUpdate();
        })()"
      >
        Set field to random number
      </button>
    </gux-button-slot>
  </fieldset>
</form>

<fieldset>
  <legend>Setting the input size</legend>
  <p>
    To set the size of an input create a class with the desired width and apply
    this class to the input element.
  </p>

  <gux-form-field-number class="width" label-position="beside">
    <input slot="input" type="number" name="e-1" value="10" />
    <label slot="label">I am long label</label>
    <span slot="error">Short error message</span>
  </gux-form-field-number>

  <gux-form-field-number label-position="beside">
    <input
      style="width: 200px"
      slot="input"
      type="number"
      name="e-1"
      value="10"
    />
    <label slot="label">I am long label</label>
    <span slot="error"
      >This is an error message with a very long text message very very
      long</span
    >
  </gux-form-field-number>

  <gux-form-field-number label-position="beside">
    <input slot="input" type="number" name="e-1" value="10" />
    <label slot="label">I am long label</label>
    <span slot="error"
      >This is an error message with a very long text message very very
      long</span
    >
  </gux-form-field-number>
</fieldset>

<style>
  .form-container {
    display: inline-flex;
    flex-direction: column;
  }

  .width::part(input-section) {
    width: 108px;
  }

  gux-form-field-number {
    margin: 10px;
  }
</style>
