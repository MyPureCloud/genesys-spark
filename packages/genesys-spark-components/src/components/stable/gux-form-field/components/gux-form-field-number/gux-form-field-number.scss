@use '../../gux-form-field.scss';
@use '../../functional-components/gux-form-field-container/gux-form-field-container.scss';
@use '../../functional-components/gux-form-field-error/gux-form-field-error.scss';
@use '../../functional-components/gux-form-field-label/gux-form-field-label.scss';
@use '../../functional-components/gux-form-field-help/gux-form-field-help.scss';

$gux-form-field-number-step-button-size: 14px;

@include gux-form-field-container.Style;
@include gux-form-field-error.Style;
@include gux-form-field-label.Style;
@include gux-form-field-help.Style;

:host {
  display: block;
}

::slotted(label) {
  @include gux-form-field.gse-ui-formControl-label;
}

::slotted(input) {
  flex: 1 1 auto;
  align-self: auto;
  order: 0;
  width: 100%;
  padding-bottom: 0;
  overflow: hidden;
  font-size: var(--gse-ui-formControl-input-contentText-fontSize);
  text-align: right;
  text-overflow: ellipsis;
  background-color: var(--gse-ui-formControl-input-backgroundColor);
  border: none;
  outline: none;

  &::placeholder {
    color: var(--gse-ui-formControl-input-placeholderColor);
    opacity: 1; //TODO : COMUI-2187
  }
}

.gux-input-and-error-container {
  flex-grow: 1;

  .gux-input {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    align-content: stretch;
    align-items: center;
    width: 100%;

    .gux-input-container {
      box-sizing: border-box;
      display: flex;
      flex: 1 1 auto;
      flex-direction: row;
      flex-wrap: nowrap;
      gap: var(--gse-ui-formControl-input-gap);
      place-content: stretch center;
      align-items: center;
      align-self: auto;
      order: 0;
      width: 100%;
      height: var(--gse-ui-formControl-input-textfield-height);
      padding: var(--gse-ui-formControl-input-padding);
      font-family: var(--gse-ui-formControl-input-contentText-fontFamily);
      font-size: var(--gse-ui-formControl-input-contentText-fontSize);
      line-height: var(--gse-ui-formControl-input-contentText-lineHeight);
      color: var(--gse-ui-formControl-input-populatedColor);
      background-color: var(--gse-ui-formControl-input-backgroundColor);
      background-image: none;
      border: var(--gse-ui-formControl-input-default-border-width)
        var(--gse-ui-formControl-input-default-border-style)
        var(--gse-ui-formControl-input-default-border-color);
      border-radius: var(--gse-ui-formControl-input-borderRadius);

      &.gux-disabled {
        opacity: var(--gse-ui-formControl-input-disabled-opacity);
      }

      &:focus-within {
        border: var(--gse-ui-formControl-input-active-border-width)
          var(--gse-ui-formControl-input-active-border-style)
          var(--gse-ui-formControl-input-active-border-color);
        border-radius: var(--gse-ui-formControl-focusRing-borderRadius);
        outline: var(--gse-ui-formControl-input-focus-border-width)
          var(--gse-ui-formControl-input-focus-border-style)
          var(--gse-ui-formControl-input-focus-border-color);
        outline-offset: var(--gse-semantic-focusRing-offset);
      }
    }

    &.gux-input-error {
      .gux-input-container {
        border-color: var(--gse-ui-formControl-input-error-border-color);
      }
    }
  }
}

.gux-step-buttons-container {
  flex: 0 1 $gux-form-field-number-step-button-size;
  align-self: auto;
  order: 0;
  margin-left: var(--gse-ui-formControl-spinButton-gap);

  .gux-step-button {
    // Flex the <gux-icon> child to ensure it doesn't have extra descender height from being inline
    display: flex;
    flex: 0 1 auto;
    align-items: center;
    align-self: auto;
    justify-content: center;
    order: 0;
    padding: 0;
    color: var(--gse-ui-formControl-input-inputIcon-defaultColor);
    background: transparent;
    border: var(--gse-ui-formControl-input-default-border-width)
      var(--gse-ui-formControl-input-default-border-style) transparent;

    &:not(:disabled):focus-visible,
    &:not(:disabled):hover {
      cursor: pointer;
      border: var(--gse-ui-formControl-input-hover-border-width)
        var(--gse-ui-formControl-input-hover-border-style)
        var(--gse-ui-formControl-input-hover-border-color);
    }

    gux-icon {
      flex: 0 0 auto;
      width: $gux-form-field-number-step-button-size;
      height: $gux-form-field-number-step-button-size;
    }
  }
}
