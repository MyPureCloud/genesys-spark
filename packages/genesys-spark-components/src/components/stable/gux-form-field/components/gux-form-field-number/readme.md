# gux-form-field-number



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description | Type                                    | Default     |
| --------------- | ---------------- | ----------- | --------------------------------------- | ----------- |
| `clearable`     | `clearable`      |             | `boolean`                               | `undefined` |
| `labelPosition` | `label-position` |             | `"above" \| "beside" \| "screenreader"` | `undefined` |


## Methods

### `guxForceUpdate() => Promise<void>`



#### Returns

Type: `Promise<void>`




## Slots

| Slot           | Description                     |
| -------------- | ------------------------------- |
| `"error"`      | Optional slot for error message |
| `"help"`       | Optional slot for help message  |
| `"input"`      | Required slot for input tag     |
| `"label"`      | Required slot for label tag     |
| `"label-info"` | Optional slot for label tooltip |


## Shadow Parts

| Part              | Description           |
| ----------------- | --------------------- |
| `"input-section"` | Style input container |


## Dependencies

### Used by

 - [gux-pagination-ellipsis-button](../../../gux-pagination/gux-pagination-buttons/gux-pagination-ellipsis-button)

### Depends on

- [gux-form-field-label-indicator](../../helper-components/gux-form-field-optional-indicator)
- [gux-form-field-input-clear-button](../../helper-components/gux-form-field-input-clear-button)
- [gux-icon](../../../gux-icon)

### Graph
```mermaid
graph TD;
  gux-form-field-number --> gux-form-field-label-indicator
  gux-form-field-number --> gux-form-field-input-clear-button
  gux-form-field-number --> gux-icon
  gux-form-field-input-clear-button --> gux-icon
  gux-pagination-ellipsis-button --> gux-form-field-number
  style gux-form-field-number fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
