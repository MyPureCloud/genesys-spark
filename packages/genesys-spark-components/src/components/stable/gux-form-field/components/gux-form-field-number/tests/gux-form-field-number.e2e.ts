import { E2EPage, newE2EPage } from '@stencil/core/testing';
import {
  newSparkE2EPage,
  a11yCheck
} from '../../../../../../test/e2eTestUtils';

import { renderConfigs } from './gux-form-field-number.common';

const axeExclusions = [];

async function newNonrandomE2EPage({
  html
}: {
  html: string;
}): Promise<E2EPage> {
  const page = await newE2EPage();

  await page.evaluateOnNewDocument(() => {
    Math.random = () => 0.5;
  });
  await page.setContent(html);
  await page.waitForChanges();

  return page;
}

describe('gux-tooltip-beta', () => {
  describe('#render', () => {
    renderConfigs.forEach(({ html, description }) => {
      it(description, async () => {
        const snapshotPage = await newNonrandomE2EPage({ html });
        const element = await snapshotPage.find('gux-tooltip-beta');

        expect(element.outerHTML).toMatchSnapshot();

        const accessibilityPage = await newSparkE2EPage({ html });

        await a11yCheck(accessibilityPage, axeExclusions);
      });
    });
  });
});
