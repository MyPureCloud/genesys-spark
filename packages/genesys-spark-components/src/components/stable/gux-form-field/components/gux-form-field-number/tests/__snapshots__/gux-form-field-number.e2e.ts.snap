// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-form-field-number #render clearable should render component as expected (1) 1`] = `"<gux-form-field-number lang="en" hydrated=""> <input slot="input" type="number" value="10" id="gux-form-field-input-i"> <label slot="label" for="gux-form-field-input-i">Label</label> </gux-form-field-number>"`;

exports[`gux-form-field-number #render clearable should render component as expected (1) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input" part="input-section">
      <div class="gux-input-container">
        <slot name="input"></slot>
      </div>
      <div class="gux-step-buttons-container">
        <button class="gux-step-button" tabindex="-1" title="Increment" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-up-small-regular" size="inherit"></gux-icon>
        </button>
        <button class="gux-step-button" tabindex="-1" title="Decrement" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-down-small-regular" size="inherit"></gux-icon>
        </button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="fa/hexagon-exclamation-solid" size="inherit"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
    <div class="gux-form-field-help">
      <div class="gux-message">
        <slot name="help"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-number #render clearable should render component as expected (2) 1`] = `"<gux-form-field-number lang="en" clearable="" hydrated=""> <input slot="input" type="number" value="10" id="gux-form-field-input-i"> <label slot="label" for="gux-form-field-input-i">Label</label> </gux-form-field-number>"`;

exports[`gux-form-field-number #render clearable should render component as expected (2) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input" part="input-section">
      <div class="gux-clear gux-input-container">
        <slot name="input"></slot>
        <gux-form-field-input-clear-button hydrated=""></gux-form-field-input-clear-button>
      </div>
      <div class="gux-step-buttons-container">
        <button class="gux-step-button" tabindex="-1" title="Increment" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-up-small-regular" size="inherit"></gux-icon>
        </button>
        <button class="gux-step-button" tabindex="-1" title="Decrement" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-down-small-regular" size="inherit"></gux-icon>
        </button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="fa/hexagon-exclamation-solid" size="inherit"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
    <div class="gux-form-field-help">
      <div class="gux-message">
        <slot name="help"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-number #render clearable should render component as expected (3) 1`] = `"<gux-form-field-number lang="en" clearable="true" hydrated=""> <input slot="input" type="number" value="10" id="gux-form-field-input-i"> <label slot="label" for="gux-form-field-input-i">Label</label> </gux-form-field-number>"`;

exports[`gux-form-field-number #render clearable should render component as expected (3) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input" part="input-section">
      <div class="gux-clear gux-input-container">
        <slot name="input"></slot>
        <gux-form-field-input-clear-button hydrated=""></gux-form-field-input-clear-button>
      </div>
      <div class="gux-step-buttons-container">
        <button class="gux-step-button" tabindex="-1" title="Increment" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-up-small-regular" size="inherit"></gux-icon>
        </button>
        <button class="gux-step-button" tabindex="-1" title="Decrement" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-down-small-regular" size="inherit"></gux-icon>
        </button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="fa/hexagon-exclamation-solid" size="inherit"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
    <div class="gux-form-field-help">
      <div class="gux-message">
        <slot name="help"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-number #render clearable should render component as expected (4) 1`] = `"<gux-form-field-number lang="en" clearable="false" hydrated=""> <input slot="input" type="number" value="10" id="gux-form-field-input-i"> <label slot="label" for="gux-form-field-input-i">Label</label> </gux-form-field-number>"`;

exports[`gux-form-field-number #render clearable should render component as expected (4) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input" part="input-section">
      <div class="gux-input-container">
        <slot name="input"></slot>
      </div>
      <div class="gux-step-buttons-container">
        <button class="gux-step-button" tabindex="-1" title="Increment" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-up-small-regular" size="inherit"></gux-icon>
        </button>
        <button class="gux-step-button" tabindex="-1" title="Decrement" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-down-small-regular" size="inherit"></gux-icon>
        </button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="fa/hexagon-exclamation-solid" size="inherit"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
    <div class="gux-form-field-help">
      <div class="gux-message">
        <slot name="help"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-number #render clearable should render component as expected (5) 1`] = `"<gux-form-field-number lang="en" clearable="" hydrated=""> <input slot="input" type="number" value="10" disabled="" id="gux-form-field-input-i"> <label slot="label" for="gux-form-field-input-i">Label</label> </gux-form-field-number>"`;

exports[`gux-form-field-number #render clearable should render component as expected (5) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input" part="input-section">
      <div class="gux-disabled gux-input-container">
        <slot name="input"></slot>
      </div>
      <div class="gux-step-buttons-container">
        <button class="gux-step-button" disabled="" tabindex="-1" title="Increment" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-up-small-regular" size="inherit"></gux-icon>
        </button>
        <button class="gux-step-button" disabled="" tabindex="-1" title="Decrement" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-down-small-regular" size="inherit"></gux-icon>
        </button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="fa/hexagon-exclamation-solid" size="inherit"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
    <div class="gux-form-field-help">
      <div class="gux-message">
        <slot name="help"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-number #render help should render component as expected 1`] = `"<gux-form-field-number hydrated=""> <input slot="input" type="number" name="e-1" value="10" id="gux-form-field-input-i" aria-describedby="gux-form-field-help-i"> <label slot="label" for="gux-form-field-input-i">Default</label> <span slot="help" id="gux-form-field-help-i">This is a help message</span> </gux-form-field-number>"`;

exports[`gux-form-field-number #render help should render component as expected 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input" part="input-section">
      <div class="gux-input-container">
        <slot name="input"></slot>
      </div>
      <div class="gux-step-buttons-container">
        <button class="gux-step-button" tabindex="-1" title="Increment" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-up-small-regular" size="inherit"></gux-icon>
        </button>
        <button class="gux-step-button" tabindex="-1" title="Decrement" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-down-small-regular" size="inherit"></gux-icon>
        </button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="fa/hexagon-exclamation-solid" size="inherit"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
    <div class="gux-form-field-help gux-show">
      <div class="gux-message">
        <slot name="help"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-number #render input attributes should render component as expected (1) 1`] = `"<gux-form-field-number lang="en" hydrated=""> <input slot="input" type="number" value="10" id="gux-form-field-input-i"> <label slot="label" for="gux-form-field-input-i">Label</label> </gux-form-field-number>"`;

exports[`gux-form-field-number #render input attributes should render component as expected (1) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input" part="input-section">
      <div class="gux-input-container">
        <slot name="input"></slot>
      </div>
      <div class="gux-step-buttons-container">
        <button class="gux-step-button" tabindex="-1" title="Increment" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-up-small-regular" size="inherit"></gux-icon>
        </button>
        <button class="gux-step-button" tabindex="-1" title="Decrement" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-down-small-regular" size="inherit"></gux-icon>
        </button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="fa/hexagon-exclamation-solid" size="inherit"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
    <div class="gux-form-field-help">
      <div class="gux-message">
        <slot name="help"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-number #render input attributes should render component as expected (2) 1`] = `"<gux-form-field-number lang="en" hydrated=""> <input slot="input" type="number" value="10" disabled="" id="gux-form-field-input-i"> <label slot="label" for="gux-form-field-input-i">Label</label> </gux-form-field-number>"`;

exports[`gux-form-field-number #render input attributes should render component as expected (2) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input" part="input-section">
      <div class="gux-disabled gux-input-container">
        <slot name="input"></slot>
      </div>
      <div class="gux-step-buttons-container">
        <button class="gux-step-button" disabled="" tabindex="-1" title="Increment" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-up-small-regular" size="inherit"></gux-icon>
        </button>
        <button class="gux-step-button" disabled="" tabindex="-1" title="Decrement" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-down-small-regular" size="inherit"></gux-icon>
        </button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="fa/hexagon-exclamation-solid" size="inherit"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
    <div class="gux-form-field-help">
      <div class="gux-message">
        <slot name="help"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-number #render input attributes should render component as expected (3) 1`] = `"<gux-form-field-number lang="en" hydrated=""> <input slot="input" type="number" value="10" required="" id="gux-form-field-input-i"> <label slot="label" for="gux-form-field-input-i">Label</label> </gux-form-field-number>"`;

exports[`gux-form-field-number #render input attributes should render component as expected (3) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label gux-required">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input" part="input-section">
      <div class="gux-input-container">
        <slot name="input"></slot>
      </div>
      <div class="gux-step-buttons-container">
        <button class="gux-step-button" tabindex="-1" title="Increment" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-up-small-regular" size="inherit"></gux-icon>
        </button>
        <button class="gux-step-button" tabindex="-1" title="Decrement" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-down-small-regular" size="inherit"></gux-icon>
        </button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="fa/hexagon-exclamation-solid" size="inherit"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
    <div class="gux-form-field-help">
      <div class="gux-message">
        <slot name="help"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-number #render label-position should render component as expected (1) 1`] = `"<gux-form-field-number lang="en" hydrated=""> <input slot="input" type="number" value="10" id="gux-form-field-input-i"> <label slot="label" for="gux-form-field-input-i">Label</label> </gux-form-field-number>"`;

exports[`gux-form-field-number #render label-position should render component as expected (1) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input" part="input-section">
      <div class="gux-input-container">
        <slot name="input"></slot>
      </div>
      <div class="gux-step-buttons-container">
        <button class="gux-step-button" tabindex="-1" title="Increment" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-up-small-regular" size="inherit"></gux-icon>
        </button>
        <button class="gux-step-button" tabindex="-1" title="Decrement" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-down-small-regular" size="inherit"></gux-icon>
        </button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="fa/hexagon-exclamation-solid" size="inherit"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
    <div class="gux-form-field-help">
      <div class="gux-message">
        <slot name="help"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-number #render label-position should render component as expected (2) 1`] = `"<gux-form-field-number lang="en" label-position="above" hydrated=""> <input slot="input" type="number" value="10" id="gux-form-field-input-i"> <label slot="label" for="gux-form-field-input-i">Label</label> </gux-form-field-number>"`;

exports[`gux-form-field-number #render label-position should render component as expected (2) 2`] = `
<div class="gux-above gux-form-field-container">
  <div class="gux-above gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input" part="input-section">
      <div class="gux-input-container">
        <slot name="input"></slot>
      </div>
      <div class="gux-step-buttons-container">
        <button class="gux-step-button" tabindex="-1" title="Increment" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-up-small-regular" size="inherit"></gux-icon>
        </button>
        <button class="gux-step-button" tabindex="-1" title="Decrement" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-down-small-regular" size="inherit"></gux-icon>
        </button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="fa/hexagon-exclamation-solid" size="inherit"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
    <div class="gux-form-field-help">
      <div class="gux-message">
        <slot name="help"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-number #render label-position should render component as expected (3) 1`] = `"<gux-form-field-number lang="en" label-position="beside" hydrated=""> <input slot="input" type="number" value="10" id="gux-form-field-input-i"> <label slot="label" for="gux-form-field-input-i">Label</label> </gux-form-field-number>"`;

exports[`gux-form-field-number #render label-position should render component as expected (3) 2`] = `
<div class="gux-beside gux-form-field-container">
  <div class="gux-beside gux-form-field-label">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input" part="input-section">
      <div class="gux-input-container">
        <slot name="input"></slot>
      </div>
      <div class="gux-step-buttons-container">
        <button class="gux-step-button" tabindex="-1" title="Increment" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-up-small-regular" size="inherit"></gux-icon>
        </button>
        <button class="gux-step-button" tabindex="-1" title="Decrement" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-down-small-regular" size="inherit"></gux-icon>
        </button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="fa/hexagon-exclamation-solid" size="inherit"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
    <div class="gux-form-field-help">
      <div class="gux-message">
        <slot name="help"></slot>
      </div>
    </div>
  </div>
</div>
`;

exports[`gux-form-field-number #render label-position should render component as expected (4) 1`] = `"<gux-form-field-number lang="en" label-position="screenreader" hydrated=""> <input slot="input" type="number" value="10" id="gux-form-field-input-i"> <label slot="label" for="gux-form-field-input-i">Label</label> </gux-form-field-number>"`;

exports[`gux-form-field-number #render label-position should render component as expected (4) 2`] = `
<div class="gux-form-field-container gux-screenreader">
  <div class="gux-form-field-label gux-screenreader">
    <slot name="label"></slot>
  </div>
  <div class="gux-input-and-error-container">
    <div class="gux-input" part="input-section">
      <div class="gux-input-container">
        <slot name="input"></slot>
      </div>
      <div class="gux-step-buttons-container">
        <button class="gux-step-button" tabindex="-1" title="Increment" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-up-small-regular" size="inherit"></gux-icon>
        </button>
        <button class="gux-step-button" tabindex="-1" title="Decrement" type="button">
          <gux-icon hydrated="" icon-name="custom/chevron-down-small-regular" size="inherit"></gux-icon>
        </button>
      </div>
    </div>
    <div class="gux-form-field-error">
      <gux-icon hydrated="" icon-name="fa/hexagon-exclamation-solid" size="inherit"></gux-icon>
      <div class="gux-message">
        <slot name="error"></slot>
      </div>
    </div>
    <div class="gux-form-field-help">
      <div class="gux-message">
        <slot name="help"></slot>
      </div>
    </div>
  </div>
</div>
`;
